src/server/rdb_save.cc

```diff
@@ -467,15 +467,9 @@ error_code RdbSerializer::SaveZSetObject(const PrimeValue& pv) {
     RETURN_ON_ERR(SaveLen(zs->Size()));
     std::error_code ec;
 
-    /* We save the skiplist elements from the greatest to the smallest
-     * (that's trivial since the elements are already ordered in the
-     * skiplist): this improves the load process, since the next loaded
-     * element will always be the smaller, so adding to the skiplist
-     * will always immediately stop at the head, making the insertion
-     * O(1) instead of O(log(N)). */
     const size_t total = zs->Size();
     size_t count = 0;
-    zs->Iterate(0, total, true, [&](sds ele, double score) mutable {
+    zs->Iterate(0, total, false, [&](sds ele, double score) mutable {
```

Comment:
this is the main fix, we saved in reverse  because it was good for skiplist but it is very bad for bptree that 
has a special optimization when inserting an element to the right of the tree (largest)

Reply:
maybe worth adding a comment for that so that we don't risk changing it ?